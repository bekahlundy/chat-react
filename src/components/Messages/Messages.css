.Messages--container {
  height: 400px;
  margin-bottom: 20px;
  overflow-y: scroll;
}

.Messages-current-user-message-container {
  text-align: right;
}

.Messages--author {
  padding: 3px 10px;
}

.Messages--message {
  display: inline-block;
}
