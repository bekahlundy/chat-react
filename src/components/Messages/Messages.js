import React, { Component } from "react";
import "./Messages.css";

export class Messages extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { currentUser, messages } = this.props;

    if (messages.length === 0) {
      return <div className="Messages-container">no messages...</div>;
    }

    return (
      <div className="Messages--container">
        {messages.map(msg => {
          return (
            <div
              key={msg.message}
              className={
                currentUser === msg.author
                  ? "Messages-current-user-message-container"
                  : "Messages-message-container"
              }
            >
              <div className="Messages--author">
                {msg.author}:{" "}
                <div className="Messages--message">{msg.message}</div>
              </div>
            </div>
          );
        })}
      </div>
    );
  }
}
