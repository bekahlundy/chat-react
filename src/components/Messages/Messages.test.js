import React from "react";
import ReactDOM from "react-dom";
import { Messages } from "./Messages";
import { render, cleanup } from "react-testing-library";

describe("Messages", () => {

  afterEach(() => {
    cleanup();
  });

  it("renders no messages text when there are no messages", () => {
    const user = "Bekah";
    const messages = [];
    const { container } = render(<Messages currentUser={user} messages={messages} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with messages", () => {
    const user = "Bekah";
    const messages = [
      {author: "bekah", message: "Hello"},
      {author: "A dog", message: "woof"}
    ]
    const { container } = render(<Messages currentUser={user} messages={messages} />);

    expect(container.firstChild).toMatchSnapshot();
  });

});
