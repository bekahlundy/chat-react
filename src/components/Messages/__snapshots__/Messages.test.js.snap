// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages renders 1`] = `
<div
  class="Messages-container"
>
  no messages...
</div>
`;

exports[`Messages renders with messages 1`] = `
<div
  class="Messages--container"
>
  <div
    class="Messages-message-container"
  >
    <div
      class="Messages--author"
    >
      bekah
      :
       
      <div
        class="Messages--message"
      >
        Hello
      </div>
    </div>
  </div>
  <div
    class="Messages-message-container"
  >
    <div
      class="Messages--author"
    >
      A dog
      :
       
      <div
        class="Messages--message"
      >
        woof
      </div>
    </div>
  </div>
</div>
`;
